Lib/fontTools/feaLib/builder.py

```diff
@@ -180,8 +197,14 @@ def build(self, tables=None):
             elif "BASE" in self.font:
                 del self.font["BASE"]
 
+    def format_location_(self, location):
+        if not location:
+            return None
+        path, line, column = location
+        return f"{path}:{line}:{column}"
+
     def get_chained_lookup_(self, location, builder_class):
-        result = builder_class(self.font, location)
+        result = builder_class(self.font, self.format_location_(location))
```

Comment:
now that I look at this, I am thinking it is kind of wasteful to pre-format the location tuple as `file:line:col` string, considering it won't be used most of the time unless an error occurs..
On the other hand, we wouldn't want to hard-code the meaning of "location"  to a text file with lines/columns for otlLib builders.
How about the `OpenTypeLibError.__str__()` checks if `location` is a tuple, then it does `":".join(str(s) for s in location)` (without assuming it's going to be three strings); if `location` is None, it skips it; and finally, if `location` is a str already, it uses it as the prefix (the latter may or may not be necessary, as one can always pass a tuple with a single str item).

Reply:
I'm going to take a slightly different but equivalent approach. We have established that users (e.g. feaLib.builder) are responsible for catching OpenTypeLibError exceptions and re-throwing them in ways that make sense to their end-users. So ordinarily the formatting the message (including the location) should be done in the class that uses otlLib. If the end-user sees an OpenTypeLibError message, that's a bug where the calling class failed to catch the exception. What I'm going to do in OpenTypeLibError is just have it try and stringify the `location` as a "best effort" and prefix it, and if it can't stringify it, just return the message without the exception.

Reply:
Except that won't work because of this:

```
    def add_subtable_break(self, location):
        log.warning(OpenTypeLibError(
            'unsupported "subtable" statement for lookup type',
            location
        ))
```

Aaargh.

Reply:
I'm going to use a location-formatter lambda attribute, and feel bad about myself.